src/library/pia/pia-utils.ts

```diff
@@ -0,0 +1,126 @@
+import * as PiaTypes from "./pia-types";
+import dayjs from "dayjs";
+
+export function piaSubstr(str: string, start: number, end: number): string {
+  let start_pos = start - 1;
+  let str_len = end - start + 1;
```

Comment:
Are these snake-cased on purpose? Seems like they should be `startPos` and `strLen`